### PR TITLE
[IDL Comm] Improve IDL for consistent naming

### PR DESCRIPTION
**Pull request by chadwalters**
Conform to predominant IDL naming convention. Also pass service and method
names as separate strings. Minor fixups on empty string usage in C#.

### CODE REVIEW

File: cs/test/comm/Epoxy/EpoxyProtocolTests.cs [hunk 1]
```diff
@@ -19,33 +19,38 @@ class EpoxyProtocolTests
     {
         private const uint GoodRequestId = 1;
         private const uint GoodResponseId = 1;
+        private const string GoodService = "MyService";
         private const string GoodMethod = "ShaveYaks";
         private const ProtocolErrorCode MeaninglessErrorCode = ProtocolErrorCode.GENERIC_ERROR;
         private static readonly IMessage<Dummy> meaninglessPayload = new Message<Dummy>(new Dummy());
-        private static readonly IMessage meaninglessError = Message.FromError(new InternalServerError() { error_code = (int)ErrorCode.InternalServerError, message = "Meaningless message"});
+        private static readonly IMessage meaninglessError = Message.FromError(new InternalServerError() { error_code = (int)ErrorCode.INTERNAL_SERVER_ERROR, message = "Meaningless message"});
```
Comment (chwarr) [inline cs/test/comm/Epoxy/EpoxyProtocolTests.cs]: Minor: `( )` not needed here.

File: doc/src/bond_comm_epoxy_wire.md [hunk 2]
```diff
@@ -346,17 +347,24 @@ connection for over 580 years before exhausting its conversation IDs. In
...
 encountered.
 
-### Method name ###
+### Service name ###
+
+The `serice_name` field is the name of the service that is being invoked.
+
+The `service_name` field must be set to a UTF-8 encoded string (without a
+BOM) that is the fully-qualified service name. Namespace elements are separated
+by the period character ('.', ASCII 0x2E). May be empty string if the message
```
Comment (chwarr) [inline doc/src/bond_comm_epoxy_wire.md]: Minor: here and below, the phrasing "may be _the_ empty string" or "may be _an_ empty string" both read better to me.

File: cs/test/comm/Epoxy/EpoxyProtocolTests.cs (hunk 1 above)
Comment (chadwalters) [inline cs/test/comm/Epoxy/EpoxyProtocolTests.cs]: The previous version had the parens as well. Can we do a separate pass to fix things like that up?

**Comment by chwarr** [comment]
:+1: LGTM
After CI builds pass, feel free to commit. Or, if you want, I can merge too.

File: cs/test/comm/Epoxy/EpoxyProtocolTests.cs (hunk 1 above)
Comment (chwarr) [inline cs/test/comm/Epoxy/EpoxyProtocolTests.cs]: Sure a separate pass is fine, or we can ignore it.
I'll probably still comment on it every time it see it, though. :-)

File: doc/src/bond_comm_epoxy_wire.md (hunk 2 above)
Comment (lalo) [inline doc/src/bond_comm_epoxy_wire.md]: there is also a typo on line 352

**Pull request merged**